app/api/domains/api.py

```diff
@@ -108,6 +109,38 @@ def format_map_basic(m: Beatmap) -> dict[str, object]:
     }
 
 
+@router.get("/search_players")
+async def api_search_players(
+    search: Optional[str] = Query(None, alias="q", min=2, max=32),
+    db_conn: databases.core.Connection = Depends(acquire_db_conn),
+):
+    """Search for users on the server by name."""
+
+    # execute the query using the database connection
+    rows = await db_conn.fetch_all(
+        "SELECT id, name FROM users WHERE name LIKE :search AND priv & 3 = 3 ORDER BY id ASC",
+        {"search": f"%{search}%"},
```

Comment:
not sure this handles the None case correctly at the moment - i think we'd want 
```suggestion
        {"search": f"%{search}%" if search is not None else None},
```

Reply:
if you could test this, would be epic

Reply:
maybe we can just disallow empty query; might be simpler